File: 75-sort-colors/sort-colors.js
var sortColors = function(nums) {
    let left = 0 , right = nums.length-1;
    for(let i=0;i<=right;){
        if(nums[i]==0){
            nums = swapNumbers(nums,left,i);
            left++;
            i++;
        }
        else if(nums[i]==2){
            nums = swapNumbers(nums,right,i);
            right--;
        }
        else{
            i++;
        }
    }
    return nums;
};

const swapNumbers = (nums,i,j) => {
    let temp = nums[i];
    nums[i] = nums[j];
    nums[j] = temp;

    return nums;
}

File: 75-sort-colors/sort-colors.java
class Solution {
    public void sortColors(int[] nums) {
        int left = 0,right = nums.length-1;
        for(int i=0;i<=right;){
            if(nums[i]==0){
                nums[i] = nums[left];
                nums[left++] = 0;
                i++;
            }else if(nums[i]==2){
                nums[i] = nums[right];
                nums[right--] = 2;
            }else{
                i++;
            }
        }
    }
}